File: javaLegacyApp/src/main/java/com/hamdan/forzenbook/java/legacy/app/MainActivity.java
package com.hamdan.forzenbook.java.legacy.app;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.hamdan.forzenbook.java.createaccount.core.viewmodel.JavaCreateAccountViewModel;
import com.hamdan.forzenbook.java.login.core.viewmodel.JavaLoginViewModel;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class MainActivity extends AppCompatActivity {

    private JavaLoginViewModel loginTest;
    private JavaCreateAccountViewModel createTest;

    // TODO in FA-114 / 116 remove this activity and these viewmodels from it

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        loginTest = new ViewModelProvider(this).get(JavaLoginViewModel.class);
        createTest = new ViewModelProvider(this).get(JavaCreateAccountViewModel.class);
    }
}

File: applegacy/src/main/java/com/hamdan/forzenbook/applegacy/view/login/MainActivity.kt
package com.hamdan.forzenbook.applegacy.view.login

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.activity.ComponentActivity
import androidx.activity.viewModels
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import com.hamdan.forzenbook.applegacy.R
import com.hamdan.forzenbook.applegacy.viewmodels.LegacyCreateAccountViewModel
import com.hamdan.forzenbook.applegacy.viewmodels.LegacyLoginViewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.launch

// TODO REMOVE THIS ACTIVITY IN FA-104 OR FA-102
@AndroidEntryPoint
class MainActivity : ComponentActivity() {
    private val loginModel: LegacyLoginViewModel by viewModels()
    private val createAccountViewModel: LegacyCreateAccountViewModel by viewModels()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val testText = findViewById<TextView>(R.id.testText)
        testText.text = loginModel.testState.value.ex.toString()
        val testButton = findViewById<Button>(R.id.testButton).setOnClickListener {
            loginModel.testFunc()
        }
        // TODO WHEN WORKING ON FA-100 and 102 REMOVE
        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED) {
                loginModel.testState.collect { state ->
                    testText.text = state.ex.toString()
                }
            }
        }
        // TODO REMOVE WHEN WORKING ON RESPECTIVE TICKETS FA-100 and 102 respectively
        loginModel.test()
    }
}
